### ISSUE TITLE
Comments on the main branch code and files

### ISSUE THREAD

**Issue by facuxpalacio** [facuxpalacio/FD-protocol-shiny-app#1]
The code as you will see is pretty rough. My main idea was to set a draft as follows:

- Use different tabs for each step (8 tabs)
- The first tab is just the description of the protocol and link to the (future) paper
- Each tab has checkboxes for the user to check what has done or not (along headings briefly explaining what the step refers to).
- I thought to develop steps 5 and 6 a bit more to include data summaries, plots, FD metrics, etc. I'm not sure if the remaining steps are worth to include anything useful, but ideas are welcome.
- The text throughout the app is intended to be just a draft, we can refine it when the app comes to an end. 

I've also uploaded two datasets, one is the classical "Iris" data and the other belongs to one of my publications. I included Iris because it has several measurements per species, so it can be thought as a trait dataset accounting for intraspecific variation. 

Awaiting for your comments!

**Comment by emmajhudgins** [comment]
Hi Facundo,

This is a great start! I think making use of Iris data is a great idea. I only made a few small grammatical changes to the code in a branch. For me, the check boxes disappear when I click them, but I assume this is a work in progress.

I'd suggest we make this repo consistent with the reproducible code+data box I started drafting for part 8 (but you can edit this box to conform to whatever you think the minimum requirements are). https://docs.google.com/document/d/1tq-VYz9vNxLWMBkaoWyX8xfO9yqo1cK8IARNZnq_GRI/edit

If you agree, you could set up the file structure to include an 'data', 'output', and 'R' subfolder, and we can detail the steps required to use all of the files in the readme. I can do a first pass at this if you'd like.

@graco-roza might know whether keeping the shiny app in one script vs. separating ui and server components is best (I've seen arguments for the latter but the app is fairly simple so it might be ok).

I think it could be nice to convert parts of this checklist to a fillable form that people can download from the shiny app, and then upload to their own repos as proof they followed the steps. I am not skilled enough in shiny at this point to immediately know how to do this yet, but I can try to learn! This looks like a good starting point https://deanattali.com/2015/06/14/mimicking-google-form-shiny/

We could examples to the fillable form for steps 1-2 based on the iris or bird data. The iris/bird answers could be shown in the form before it gets filled as default answers.

For step 1: The current MS draft actually makes having a specific research question optional, so it could be good to have an initial decision tree where people decide if they're going exploratory or question-driven. I think it could be cool to show examples of each type of question with iris and bird data. 

step2: people could fill in their preregistration doi/other identifier if they check yes and provide the location of the power analysis (maybe in the focal repo).

step 3-4
- Could we have people upload their files at these stages rather than step 5? 
- Could we make some dummy community data here? And then we could show the output of 'head' on the resulting data frames n both steps.
- The fillable form could include file paths to these data sources

step 4

- should we ask about accounting for phylogenetic structure here too? 
- I think brief definitions of things like soft/hard traits could mean that people don't have to keep referring to the MS.

step5 
- This one is super impressive!!! 
- maybe we can reorder the left panel so that the things that happen before plotting are listed first (e.g. accounting for missingness/bias).
- The need to deal with multicollinearity and the need for transformation requirements could be shown visually - maybe we can have a tab for transformations that replot the histograms after some standard transformations. Maybe we can also discuss outliers here? e.g. having another tab for outlier detection?
- It also seems like the collinearity box is being printed far down the page
- people could fill in any transformations/outliers/multicollinearity fixes they had to do in the fillable form

step6 
- again, here I think we can have some example answers from the example dataset in a fillable form
- i'm not sure what the subetting is in "Did you subset your trait data?" - maybe give a few examples

step 7 - could we provide examples of some validation types in tabs using the iris or bird data? e.g. jackknifing and permutation tests? I can help with this

step 8 - maybe we can integrate the reproducible code box into this step more fully 

Let me know if you want me to expand on anything!

Emma

**Comment by facuxpalacio** [comment]
Hi Emma, thank you so much for your nice feedback, you've made me think about manys stuffs! 

I'll do my best to follow your box on reproducible code+data, but feel free to indicate if I'm getting this wrong or missing something. 

It is weird that check boxes disappear when you click them, it shouldn't happen, or at least, this never has happened to me. We should check this out.

As the app is pretty simple right now, I don't think we need to separate the ui and the server.

Step 1:

- Sounds good. Do you have an idea of what kind of input should we use? You can take a look at the available inputs in the shiny cheatsheet https://shiny.rstudio.com/images/shiny-cheatsheet.pdf

Step 2:

- Mmmh, I should figure out how to do this! Could we simulate this with fake data or something similar?

Step 3:

- I try to imagine how I'd use the app for my own analyses. Steps 3 and 4 are intended to check whether you collected, and how you collected the data. Some information could be directly retrieved when uploading a file (e.g. number of traits, types of traits), but this may become tricky as people upload their data in multiple forms with columns that may have nothing to do with the analysis. Besides, I think it would be best for the user to upload their files and see the dataset in the same output, rather than switching between files and output tabs.
- Thinking in your comment on a final report (which is a very cool idea), then we should move to textInput so that people can fill entries that will appear in the report. Or should se keep both, so people can check the boxes and add their own info?
- I agree to include a community matrix to the "data" subfolder ( I was going to include the corresponding bird community dataset). Indeed, I 've included a new file browser in Step 5, as well as a heading output.

Step 4

- If we include an entry for phylogenetic information, we should add it to the manuscript.

Step 5

- OK, I'll add your changes. This step and Step 6 are maybe the harder ones (but also the funniest)t o code, as they have many reactive programming.

Step 6
- Good. Subsetting means to group set of traits based on their ecological meaning (e.g., you have 10 traits but you compute FD using 3 only traits accounting for pollen dispersal).

Step 7

- Sure! It would be great if you can take on this.

Step 8

- I agree to include the reproducible code box here.

To summarize, Emma could you take on:

-

**Comment by facuxpalacio** [comment]
@emmajhudgins I've created a data folder. Before going on, I've got a few questions:

- Should I upload the app.R file in the R folder? If so, does the main branch remain unaltered?
- What should I put in the the output folder? If I understand correctly, I cannot create an empty folder.

Thanks in advance

Facu

**Comment by facuxpalacio** [comment]
@emmajhudgins maybe you're right to upload the community and trait datasets in Steps 3 and 4, respectively. I've put two file browsers in the "Dataset tab" in Step 5, and it looks a bit overloaded once both datasets appear.
